src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/Application/DataLock/RunDataLockValidationQuery/RunDataLockValidationQueryHandler.cs

```diff
@@ -239,62 +213,65 @@ private Commitment.Commitment GetMatchingCommitment(DateTime date, Commitment.Co
+                                                        Commitment.Commitment[] commitments, 
+                                                        int period, 
+                                                        DateTime periodDate,
+                                                        IEnumerable<IncentiveEarnings> incentiveEarnings)
         {
             var priceEpisodePeriodMacthes = new List<PriceEpisodePeriodMatch.PriceEpisodePeriodMatch>();
 
-            var allEarningsPayable = commitment.PaymentStatus == (int)PaymentStatus.Active || commitment.PaymentStatus == (int)PaymentStatus.Completed;
-
-            priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, commitment, period, TransactionType.Learning, allEarningsPayable));
-            priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, commitment, period, TransactionType.Completion, allEarningsPayable));
-            priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, commitment, period, TransactionType.Balancing, allEarningsPayable));
-
-
-            priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, commitment, period, TransactionType.Balancing16To18FrameworkUplift, allEarningsPayable));
-            priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, commitment, period, TransactionType.Completion16To18FrameworkUplift, allEarningsPayable));
-            priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, commitment, period, TransactionType.OnProgramme16To18FrameworkUplift, allEarningsPayable));
-            priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, commitment, period, TransactionType.FirstDisadvantagePayment, allEarningsPayable));
-            priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, commitment, period, TransactionType.SecondDisadvantagePayment, allEarningsPayable));
+            var matchingCommitment = GetMatchingCommitment(periodDate, commitments);
+            if (matchingCommitment != null)
+            {
+                var allEarningsPayable = matchingCommitment.PaymentStatus == (int)PaymentStatus.Active || matchingCommitment.PaymentStatus == (int)PaymentStatus.Completed;
+                priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, matchingCommitment, period, TransactionTypesFlag.AllLearning, allEarningsPayable));
+            }
 
-            priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, commitment, period, TransactionType.BalancingMathsAndEnglish, allEarningsPayable));
 
-            priceEpisodePeriodMacthes.Add(GetPriceEpisodePeriodMatch(priceEpisode, commitment, period, TransactionType.OnProgrammeMathsAndEnglish, allEarningsPayable));
+            if (priceEpisode.FirstAdditionalPaymentThresholdDate.HasValue && incentiveEarnings.Any(x=> x.Period == period && x.First16To18EmployerIncentive !=0))
```

Comment:
the if conditions that are then added to the result set are almost identical - consider refactoring